src/EffectWatcher.re

```diff
@@ -0,0 +1,35 @@
+type context = {revEffects: ref(list(Effect.t))};
+
+module Internal = {
+  let unhandledEffectEvent = Event.create();
+  let currentContext: ref(option(context)) = ref(None);
+};
```

Comment:
The layers of `ref` here seem unnecessarily convoluted. And since it's stored in a record it would generally be better to use a `mutable`, as that's less indirection and allocation, and because the `ref` then can't be passed around independently of the record that owns it.

It could also be better encapsulated, although the scope here is pretty small anyway.

Also seems like a stack is the more appropriate data structure to use here. A list and a stack is structurally equivalent though, but I think using the name `effectStack` rather than `revEffects` better communicates that it's used as a LIFO data structure here.

Reply:
Good point, I don't think it's necessary to have both layers - I streamlined it so we only have one layer (at `currentContext`).